qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2017 IBM RESEARCH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+"""IBMQJob module
+
+This module is used for creating asynchronous job objects for the
+IBM Q Experience.
+"""
+
+from concurrent import futures
+import logging
+
+from qiskit.backends import BaseJob
+from qiskit.backends.basejob import JobStatus
+from qiskit._qiskiterror import QISKitError
+from qiskit._result import Result
+
+logger = logging.getLogger(__name__)
+
+
+class IBMQJob(BaseJob):
+    """IBM Q Job class
+
+    Attributes:
+        _executor (futures.Executor): executor to handle asynchronous jobs
+    """
+    _executor = futures.ThreadPoolExecutor()
+
+    def __init__(self, fn, q_job, api, timeout, submit_info):
```

Comment:
Actually what you've described is sort of how the IBMQ backend works.  For the IBMQJob object what really happens during initialization is 'start using the supplied job_id to monitor the job'. Actual starting of the job is handled in the ibmqbackend._submit method. This was necessary so the backend could get a job_id and initialize the job instance with it so it could be available to the user.

Reply:
Can you at least update the docstring of the base class, for consistency?

Reply:
yes, I corrected that. Thanks!